<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="base_frame" default="base_footprint" />
  <arg name="target_frame" default="map" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" >
  
  <!-- Local planner -->
  <!-- <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" /> -->
   <param name ="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /> 
  <!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" /> -->
  
  <!-- Global Planner -->
  <!-- <param name="base_global_planner" value="RRTstar_planner/RRTstarPlannerROS" /> -->
  <!--  <param name="base_global_planner" value="aco_ros/aco_planner_v2" /> -->
    <param name="base_global_planner" value="global_planner/GlobalPlanner" /> 
	
   <rosparam file="$(find ant_colony_path_planner)/params/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ant_colony_path_planner)/params/costmap_common.yaml" command="load" ns="local_costmap" /> 
  <rosparam file="$(find ant_colony_path_planner)/params/costmap_local.yaml" command="load"/> 
  <rosparam file="$(find ant_colony_path_planner)/params/costmap_global.yaml" command="load"/>
    
    <rosparam file="$(find ant_colony_path_planner)/params/move_base_params.yaml" command="load" /> 
     
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_$(arg model).yaml" command="load" /> 
    
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" /> 
    
     <rosparam file="$(find ant_colony_path_planner)/params/global_planner_params.yaml" command="load" />
  </node>
</launch>
